### DISCUSSION TITLE
How to get input tensor shape in onnxruntime-web?

### DISCUSSION THREAD

**Discussion by wirhabenzeit** [microsoft/onnxruntime#17682] [General]
Given an `ort` session, say 
```javascript
session = await ort.InferenceSession.create(
  "model.onnx", { executionProviders: ["wasm"] }
)
```
how can I get the expected input shape? 

To get the input names, I can do `session.inputNames` but the docs don't explain how to get the input shapes.

**Comment by hariharans29** [comment]
You can inspect the model by opening it on Netron and checking what the model expcts its input shape to be

**Reply by wirhabenzeit** [reply]
My question was about the Javascript-runtime specifically. My use case is that I load one of several `model.onnx` file and depending on which one I downscale the data in Javascript before feeding it into `onnxruntime`. Now I can of course somewhere store the information on the model input shape but it seems superfluous since this information is stored in the model itself. For instance the python API exposes such information using `session.get_inputs()` but I take it the Javascript runtime does not?

**Reply by hariharans29** [reply]
I see. @fs-eire / @guschmue - Does onnxruntime-web allow querying the model metadata like input shape ?

**Reply by fs-eire** [reply]
we don't support this feature so far.

The major reason is that for most of the use cases we expect users to know the input type and shape requirement, so we lack motivation to implement this feature.

There are also other trivial reasons -
- some customers want to optimize for artifact size, exporting metadata increases that for some and cannot be reduced using tree-shaking (because it crosses JS/WASM boundrary).
- implementation is non-trivial as the JS API also for node.js binding and react-native.

If you are interested in it, contribution is welcome.
